### PR TITLE
Architecture Embed Group

### LINKED ISSUE #12: Centralize the update of player chat embed
Player function next, last, stop all have the same code to update the player (and all child gui's) after a player state change. This could be simplified to a function.

Other functions in the class could also benefit from this simplification, reducing simple bugs.

https://github.com/borisliao/mbox/blob/674f1eaf3cc4a0d0607239aa69bfb49eb1a930cf/src/music/player.py#L133

### LINKED ISSUE #51: Bot shows stale data when you use slash command
**Describe the bug**
If you use /pause in the command channel after the song is done it shows the last played video in the player gui.

**To Reproduce**
1. Play a song
2. End the song
3. type /pause in the command channel

**Expected behavior**
It should ignore the slash command

### PR DESCRIPTION
**Pull request by borisliao**
The final blocking architecture change before features can start to be implemented. 
Changes:

- Implemented Factory
- Removed ChatEmbed, Cache
- Renamed MusicQueue -> Queue ; Buttons -> Reactions
- All Player functions now observe and follow the spec to use factories to generate embeds (nearly a total rewrite!)

### CODE REVIEW

**Review by AvJain — DISMISSED** [review]
Changes are fine, reviewed with @borisliao and made some minor suggestions. But those should not block him from merging.

File: src/music/player.py [hunk 1]
```diff
@@ -50,416 +35,432 @@ def __init__(self, voice_channels, ffmpeg_path, messenger: Messenger, text_chann
...
+    ########## General Helper Functions ##########
+    async def register_command_channel(self, command_channel: TextChannel):
+        """Sends and registers all embeds that this player implements to the command channel"""
+
+        self.reporter_message = await command_channel.send(embed=self.reporter_embed)
+        self.lyrics_message   = await command_channel.send(embed=self.lyrics_embed)
+        self.queue_message    = await command_channel.send(embed=self.queue_embed)
+        self.player_message   = await command_channel.send(embed=self.player_embed)
+
+    def metadata_youtube_dl(self, info: dict) -> dict[str,str]:
```
Comment (Karan-Jain) [inline src/music/player.py:427]: Move the functions below to helpers,py

File: src/commander/element/QueueEmbed.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        self.title = 'Queue'
+        self.queue: Queue = kwargs.get('queue', None)
+
+
+        if self.queue and self.queue.index != None:
+            self.description = self.get_description()
+        else:
+            self.description = f'Nothing is in your queue. {USAGE_TEXT}'
+
+    def get_description(self) -> str:
```
Comment (AvJain) [inline src/commander/element/QueueEmbed.py]: change to set_description, as that is what this method is doing

File: src/commander/EmbedFactory.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+from discord.embeds import Embed
+
+from src.commander.element.PlayerEmbed import PlayerEmbed
+from src.commander.element.QueueEmbed import QueueEmbed
+from src.commander.element.LyricsEmbed import LyricsEmbed
+from src.commander.element.ReporterEmbed import ReporterEmbed
+
+
+class EmbedFactory:
+    def create_embed(type: str, **kwargs) -> Embed:
```
Comment (Karan-Jain) [inline src/commander/EmbedFactory.py]: Change  type to embed_type, since type shadows a builtin

File: src/music/player.py (hunk 1 above)
Comment (borisliao) [reply src/music/player.py:427]: These types of changes are now in issue #73. It won't be done in this PR.

**Review by Karan-Jain — APPROVED** [review]
looks good!

**Review by AvJain — APPROVED** [review]

**Pull request merged**